### PR TITLE
*: add createCRD flag to etcd restore operator

### PR DESCRIPTION
**Pull request by fanminshi**
add createCRD flag allowing user to disable etcd restore operator on creating EtcdRestore CRD automatically.

### CODE REVIEW

File: cmd/restore-operator/main.go [hunk 1]
```diff
@@ -40,8 +41,13 @@ var (
 	// This is the address of k8s service to restore operator itself for accessing
 	// backup HTTP endpoints. For example, "restore-operator:19999"
 	serviceAddrForSelf string
+	createCRD          bool
 )
 
+func init() {
+	flag.BoolVar(&createCRD, "create-crd", true, "The restore operator will not create the EtcdRestore CRD when this flag is set to false.")
```
Comment (hongchaodeng) [inline cmd/restore-operator/main.go]: Doesn't it need to call `flag.Parse()`?

**Comment by hongchaodeng** [comment]
Please also fix flag.Parse() for backup operator in this PR: https://github.com/coreos/etcd-operator/pull/1781#issuecomment-352129845

File: cmd/restore-operator/main.go (hunk 1 above)
Comment (fanminshi) [reply cmd/restore-operator/main.go]: yeah, that's the issue.

**Comment by fanminshi** [comment]
Manual Test:

```yaml
apiVersion: extensions/v1beta1
kind: Deployment
metadata:
  name: etcd-restore-operator
  labels:
    name: etcd-restore-operator
spec:
  replicas: 1
  template:
    metadata:
      labels:
        name: etcd-restore-operator
    spec:
      containers:
      - name: etcd-restore-operator
        image: gcr.io/coreos-k8s-scale-testing/etcd-operator:fanmin
        imagePullPolicy: Always
        command: ["etcd-restore-operator", "-create-crd=false"]
        env:
        - name: MY_POD_NAMESPACE
          valueFrom:
            fieldRef:
              fieldPath: metadata.namespace
        - name: MY_POD_NAME
          valueFrom:
            fieldRef:
              fieldPath: metadata.name
        - name: SERVICE_ADDR
          value: "etcd-restore-operator:19999"
---
apiVersion: v1
kind: Service
metadata:
  name: etcd-restore-operator
spec:
  selector:
    name: etcd-restore-operator
  ports:
  - protocol: TCP
    port: 19999
```

restore-operator output:
```sh
time="2017-12-15T22:58:23Z" level=info msg="starting restore controller" pkg=controller
E1215 22:58:23.921617       1 reflector.go:205] github.com/coreos/etcd-operator/pkg/controller/restore-operator/controller.go:47: Failed to list *v1beta2.EtcdRestore: the server could not find the requested resource (get etcdrestores.etcd.database.coreos.com)```

**Comment by hongchaodeng** [comment]
LGTM

Please also fix backup-operator.

**Comment by fanminshi** [comment]
merge when green.

**Pull request merged**